src/components/tableOfContents.tsx

```diff
@@ -1,6 +1,20 @@
 import React from "react";
 
-export default ({ toc: { items } }) => {
+type _Item<T> = {
+  title?: string;
+  url?: string;
+  items: T[];
+};
+
+interface Item extends _Item<Item> {}
```

Comment:
Oh interesting, I thought this could be normally written recursively like this https://www.typescriptlang.org/play?#code/C4TwDgpgBAKhDOwoF4oG8BQVtWACwEsA7AcwC4pEAnYkgbixwGNCAbAEyoiIrkQG0AuhgC+GDEwD2RRLgTBe8lOny0KAcgDueAIbAAbhCrqANFBYEOXHlCEigA

```tsx
type Item = {
  title?: string;
  url?: string;
  items: Item[];
};
```